sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py

```diff
@@ -109,10 +135,23 @@ def __init__(self, path: List[str], *, kind: Literal["MultiHash"] = "MultiHash",
     def __init__(self, path: str, *, kind: Literal["Hash"] = "Hash", version: int = 2) -> None:
         ...
 
+    @overload
+    def __init__(self, partition_key_definition: Union["PartitionKey", Dict[str, Any]]) -> None:
+        ...
+
     def __init__(self, *args, **kwargs):
-        path = args[0] if args else kwargs['path']
-        kind = args[1] if len(args) > 1 else kwargs.get('kind', 'Hash' if isinstance(path, str) else 'MultiHash')
-        version = args[2] if len(args) > 2 else kwargs.get('version', 2)
+        if (args and len(args) == 1 and isinstance(args[0],
+                                                   (dict, PartitionKey))) or 'partition_key_definition' in kwargs:
+            # If a single dictionary or PartitionKey object is passed, use it as the partition key definition
+            partition_key_definition = args[0] if args else kwargs.get("partition_key_definition", {})
+            path = partition_key_definition.get('path')
+            kind = partition_key_definition.get('kind', 'Hash')
+            # When Partition Key Definition is Passed, if no version is available then default to 1
+            version = partition_key_definition.get('version', 1)
```

Comment:
Earlier we were defaulting to version 2, why are we changing this to version 1?
In the constructor, we should always default the version to 2 if not passed. And when getting the response from the backend / service, we should default it to 1 if the version field is not present. These are 2 separate workflows.